spacy/training/loggers.py

```diff
@@ -26,14 +26,36 @@ def setup_table(
     return final_cols, final_widths, ["r" for _ in final_widths]
 
 
+# We cannot rename this method as it's directly imported
+# and used by external packages such as spacy-loggers.
 @registry.loggers("spacy.ConsoleLogger.v2")
 def console_logger(
     progress_bar: bool = False,
     console_output: bool = True,
     output_file: Optional[Union[str, Path]] = None,
 ):
     """The ConsoleLogger.v2 prints out training logs in the console and/or saves them to a jsonl file.
-    progress_bar (bool): Whether the logger should print the progress bar.
+    progress_bar (bool): Whether the logger should print a progress bar tracking the steps till the next evaluation pass.
+    console_output (bool): Whether the logger should print the logs on the console.
+    output_file (Optional[Union[str, Path]]): The file to save the training logs to.
+    """
+    return console_logger_v3(
+        progress_bar=None if progress_bar is False else "eval_steps",
+        console_output=console_output,
+        output_file=output_file,
+    )
+
+
+@registry.loggers("spacy.ConsoleLogger.v3")
+def console_logger_v3(
+    progress_bar: Optional[str] = None,
+    console_output: bool = True,
+    output_file: Optional[Union[str, Path]] = None,
+):
+    """The ConsoleLogger.v3 prints out training logs in the console and/or saves them to a jsonl file.
+    progress_bar (Optional[str]): Type of progress bar to show in the console. Allowed values:
+        all_steps - Tracks the number of steps until `training.max_steps` is reached.
+        eval_steps - Tracks the number of steps until `training.eval_frequency` is reached.
```

Comment:
Not too happy with the naming of either type, to be honest. Suggestions are welcome.

Reply:
Maybe it could be just called `progress` and its might be clearer like `--progress train` or `--progress eval`?

And for the explanation maybe like "train - Track the number of steps from the beginning of training until the full training run is complete (training.max_steps is reached). eval - Track the number of steps between the previous and next evaluation (training.eval_frequency is reached)".

I would just give this kind of explanation there because i personally had to double take a bit on what do these mean exactly.

Reply:
That sounds better, yeah. Thanks, will change it.